aten/src/ATen/native/mkldnn/Conv.cpp

```diff
@@ -329,7 +329,15 @@ Tensor mkldnn_convolution(
     IntArrayRef stride,
     IntArrayRef dilation,
     int64_t groups) {
-  bool use_channels_last = mkldnn_conv_use_channels_last(input_t, weight_t);
+  bool use_channels_last;
+  // Enables Channel last layout for 2D Convolution with Float datatype.
+  // Boosts performance by calling forward convolution with channel last (NHWC).
+  if (input_t.scalar_type() == at::kFloat &&  input_t.ndimension()==4 && (!(input_t.is_mkldnn() || weight_t.is_mkldnn()))) {
+      use_channels_last =true;
```

Comment:
@eqy , Thank you. 
In _mkldnn_convolution_out function that is in https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/mkldnn/Conv.cpp#L224, the input and weight are being converted to channel last as the memory format selected channel last.
`auto memory_format = mkldnn_convolution_memory_format(input_t.ndimension(), is_channels_last);
  auto input = input_t.is_mkldnn() ? input_t : input_t.contiguous(memory_format);
  auto weight = weight_t.is_mkldnn() ? weight_t : weight_t.contiguous(memory_format);`

Reply:
Thanks for the correction! However that still potentially unsafe as `.contiguous(memory_format)` would only be called if `input_t.is_mkldnn()` is true?

Reply:
Hi @eqy , here actually the code flow will be as below:
 auto input = input_t.is_mkldnn() ? input_t : input_t.contiguous(memory_format);

Here contiguous is called if the input is not mkldnn  and also in PR change it is taken care that the channel_last flag is true only input not mkldnn.
So, when input is not mkldnn,  channellast flag is set and in mkldnn_convolution_out function since input is not mkldnn it will set input as channel last because contiguous function is called with memory format (that is channel last as true)

Reply:
@eqy , Thank you for the inputs.
1.In few failing tests, its testing python test/test_modules.py test_memory_format_nn_Conv1d_cpu_float32 with different env settings like PYTORCH_TEST_WITH_CROSSREF=1, PYTORCH_TEST_WITH_ASAN=1 PYTORCH_TEST_WITH_UBSAN=1 here actually its checking memory format and its expecting nchw for output. But actually this PR change will create output with nhwc.
and 
2.few dynamo and inductor tests are failing (ex: test_retracibility_nested_list_out_dynamic_shapes)

Reply:
@akote123 sorry for the delay, just coming back to work from CNY leave.

The change in this PR is not correct. Please notice that the decision of which `memory format` to use is quite a complexed topic. And the checks in file `mkldnn/conv.cpp` is just a double insurance since `torch.mkldnn_convolution` is a function that exposes to users which can be used directly even not recommended.

The memory formats for input and weight tensor have been converted (if needed to be converted) in the `Convoluttion.cpp`: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Convolution.cpp#L1407

So if you need to accomplish your original target: you need to make your change here in the function `mkldnn_conv_use_channels_last`: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/ConvUtils.h#L380